web/crux-ui/src/components/registries/edit-registry-card.tsx

```diff
@@ -391,3 +394,79 @@ const GithubRegistryFields = (props: EditRegistryTypeProps<GithubRegistryDetails
     </>
   )
 }
+
+const GoogleRegistryFields = (props: EditRegistryTypeProps<GoogleRegistryDetails>) => {
+  const { formik } = props
+
+  const { t } = useTranslation('registries')
+
+  const uploadHandler = keyFile => {
+    const fileReader = new FileReader()
+    fileReader.readAsText(keyFile, 'UTF-8')
+    fileReader.onload = event => {
+      const json = JSON.parse(event.target.result.toString())
+      formik.setFieldValue('user', json.client_email ? json.client_email : '')
+      formik.setFieldValue('token', json.private_key ? json.private_key : '')
+    }
+  }
+
+  return (
+    <>
+      <DyoLabel className="text-light mt-2">{t('tips.google')}</DyoLabel>
+
+      <DyoInput
+        className="max-w-lg"
+        grow
+        name="url"
+        type="text"
+        label={t('url')}
+        onChange={formik.handleChange}
+        value={formik.values.url}
+        message={formik.errors.url}
+      />
+
+      <div className="mr-auto">
+        <DyoToggle
+          className="text-bright mt-8"
+          name="_private"
+          nameChecked={t('private')}
+          nameUnchecked={t('public')}
+          checked={formik.values._private}
+          setFieldValue={formik.setFieldValue}
+        />
+      </div>
+
+      {!formik.values._private ? null : (
+        <>
+          <DyoSingleFileUploadInput
```

Comment:
What does the `single` mean in the component name?

Reply:
Single should mean that you can upload only one file, but I will refactor the component to handle single or multiple file uploads.